server/src/db/migrations/20220119232016_store-front.ts

```diff
@@ -0,0 +1,123 @@
+import { Knex } from "knex";
+
+
+export async function up(knex: Knex): Promise<void> {
+
+    knex.schema.hasTable('InventoryItem').then((exists) => {
+        if (!exists) {
+            return knex.schema.createTable('InventoryItem', (t) => {
+                t.increments('id').primary();
+                t.text('image');
+                t.text('name');
+                t.text('unit');
+                t.text('pluralUnit');
+                t.integer('count');
+                t.float('pricePerUnit');
+            });
+        }
+    });
+
+    // if we know every label we can use an enum instead of this table
+    knex.schema.hasTable('Label').then((exists) => {
+        if (!exists) {
+            return knex.schema.createTable('Label', (t) => {
+                t.increments('id').primary();
+                t.text('label');
+            });
+        }
+    });
+
+    knex.schema.hasTable('InventoryItemLabel').then((exists) => {
```

Comment:
Is a differentiation between Label and InventoryItemLabel necessary?

Reply:
its so that the user can create their own labels. Using a table for the labels instead of an enum